File: U2/PAO202551_ELECTRO_BDDNUBE_RESTFULL_JAVA_GR01/04. SERVIDOR/Comercializadora_RestFul_Java_GR01/src/main/java/ec/edu/monster/DTOs/FacturaDTO.java
package ec.edu.monster.DTOs;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

/**
 * DTO para respuesta de Factura
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class FacturaDTO {
    private Integer facturaId;
    private String numeroFactura;
    private String cedulaCliente;
    private String nombreCliente;
    private String formaPago;
    private BigDecimal subtotal;
    private BigDecimal descuento;
    private BigDecimal total;
    private String numeroCredito;
    private LocalDateTime fechaEmision;
    private List<DetalleFacturaDTO> detalles;
}

File: U2/PAO202551_ELECTRO_BDDNUBE_RESTFULL_JAVA_GR01/04. CLICON/Cliente_Consola/Cliente_Consola/Models/FacturaDTO.cs
namespace Comer_CliCon_SOAP_DotNet_GR01.Models
{
    public class FacturaDTO
    {
        public int FacturaId { get; set; }
        public string NumeroFactura { get; set; } = string.Empty;
        public string CedulaCliente { get; set; } = string.Empty;
        public string NombreCliente { get; set; } = string.Empty;
        public string FormaPago { get; set; } = string.Empty;
        public decimal Subtotal { get; set; }
        public decimal Descuento { get; set; }
        public decimal Total { get; set; }
        public string NumeroCredito { get; set; } = string.Empty;
        public DateTime FechaEmision { get; set; }
        public List<DetalleFacturaDTO> Detalles { get; set; } = new();
    }

    public class DetalleFacturaDTO
    {
        public int ProductoId { get; set; }
        public string NombreProducto { get; set; } = string.Empty;
        public int Cantidad { get; set; }
        public decimal PrecioUnitario { get; set; }
        public decimal Subtotal { get; set; }
    }

    public class SolicitudFacturaDTO
    {
        public string CedulaCliente { get; set; } = string.Empty;
        public string NombreCliente { get; set; } = string.Empty;
        public string FormaPago { get; set; } = string.Empty;
        public string NumeroCredito { get; set; } = string.Empty;
        public List<ItemFacturaDTO> Items { get; set; } = new();
    }

    public class ItemFacturaDTO
    {
        public int ProductoId { get; set; }
        public int Cantidad { get; set; }
    }
}
